### PR TITLE
INT-2812: Rewrite the react self-hosted docs

### PR DESCRIPTION
**Pull request by tiny-james**
Related Ticket: 

Description of Changes:
* Split react self-hosted docs into 4 (instead of 2)
* Rewrote and tested the whole process
* Reviewed and fixed the technical reference

Note that I am aware that there is a lot of "almost duplicated" steps currently where the same codesample is used with slight modifications to paths and names. I plan to work on improving that while people review the content.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [ ] Documentation Team Lead has reviewed
- [ ] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/partials/integrations/react-quick-start.adoc [hunk 1]
```diff
@@ -79,67 +167,440 @@ export default function App() {
...
+ifeval::["{productUse}" == "host"]
+. Install the `+tinymce-react+` package and save it to your `+package.json+` with `+--save+`.
++
+[source,sh]
+----
+npm install --save @tinymce/tinymce-react
+----
 
-* [[deploying-tinymce-independent]] Deploying {productname} independent of the React application. To use a {productname} instance that has been deployed independent of the React application, use: xref:using-the-tinymcescriptsrc-prop[the `+tinymceScriptSrc+` prop], or xref:using-an-html-script-tag[an HTML `+<script>+` tag].
+. Unzip the content of the `+tinymce/js+` folder from the https://www.tiny.cloud/my-account/downloads/[{productname} zip] into the `+public+` folder. Afterwards the directory listing should be similar to below:
```
Comment (lnewson) [inline modules/ROOT/partials/integrations/react-quick-start.adoc:304]:
```suggestion
. Unzip the content of the `+tinymce/js+` folder from the link:{download-enterprise}[{productname} zip] into the `+public+` folder. Afterwards the directory listing should be similar to below:
```
We should be using the global attributes for links like this as it makes it easier to change if the website structure changes (which has happened a few times).

Comment (lnewson) [reply modules/ROOT/partials/integrations/react-quick-start.adoc:304]: Also I've just noticed when looking up the above attribute, that we also have ZIP downloads for community via https://www.tiny.cloud/get-tiny/self-hosted/. It's doubtful anyone would use that when they can use NPM though, but that might be why this wasn't specific previously 🤷

File: modules/ROOT/nav.adoc [hunk 2]
```diff
@@ -16,8 +16,10 @@
 **** xref:rails-cloud.adoc[Ruby on Rails]
 *** Self-hosted
 **** React
-***** xref:react-pm.adoc[Using a package manager]
-***** xref:react-zip.adoc[Using a .zip package]
+***** xref:react-pm-host.adoc[Using a package manager with hosting]
```
[the remark this answers is not in the document]
Comment (tiny-james) [reply modules/ROOT/nav.adoc:19]: I'll create an alias for them.

[the remark this answers is not in the document]
Comment (tiny-james) [reply modules/ROOT/nav.adoc:19]: Done

**Comment by tiny-james** [comment]
Note the branch was renamed which caused this to auto-close. The replacement PR is open [here](https://github.com/tinymce/tinymce-docs/pull/2407).

**Pull request closed without merging**